### PR TITLE
ci: add `angular-eslint` plugin

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [x] Build or CI related changes
- [ ] Documentation content changes

## What is the new behavior?

https://github.com/angular-eslint/angular-eslint

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by nsbarsukov — APPROVED** [review]

File: projects/kit/directives/unfinished-validator/unfinished-validator.directive.ts [hunk 1]
```diff
@@ -14,6 +14,9 @@ import {tuiCreateUnfinishedValidator} from '@taiga-ui/kit/validators';
     ],
 })
 export class TuiUnfinishedValidatorDirective implements Validator {
+    @Input('tuiUnfinishedValidator')
+    message: string | null = null;
```
Comment (waterplea) [inline projects/kit/directives/unfinished-validator/unfinished-validator.directive.ts:18]: It should support boolean attribute `<tui-input-date tuiUnfinishedValidator>`. Does it still work like that?

File: projects/cdk/components/alert-host/alert-host.component.ts [hunk 2]
```diff
@@ -17,6 +17,7 @@ import {Observable} from 'rxjs';
     templateUrl: './alert-host.template.html',
     styleUrls: ['./alert-host.style.less'],
     // So that we do not force OnPush on custom alerts
+    // eslint-disable-next-line @angular-eslint/prefer-on-push-component-change-detection
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/cdk/components/alert-host/alert-host.component.ts:20]: `but it's weird to have to disable a rule when you explicitly write a particular strategy 🤔` unfortunately this rule works in such a way that it prohibits the default strategy

File: projects/addon-doc/src/components/page/page.component.ts [hunk 3]
```diff
@@ -33,13 +32,15 @@ export class TuiDocPageComponent {
     @Input()
     path = '';
 
+    @Input()
+    deprecated: string | null = null;
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/addon-doc/src/components/page/page.component.ts]: fixed

File: projects/addon-doc/src/components/main/main.component.ts [hunk 4]
```diff
@@ -15,6 +16,7 @@ import {Subject} from 'rxjs';
     templateUrl: './main.template.html',
     styleUrls: ['./main.style.less'],
     encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush,
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/addon-doc/src/components/main/main.component.ts]: fixed

File: projects/kit/directives/unfinished-validator/unfinished-validator.directive.ts (hunk 1 above)
Comment (splincode) [reply projects/kit/directives/unfinished-validator/unfinished-validator.directive.ts]: fixed

File: projects/addon-editor/components/toolbar-tools/table-size-selector/table-size-selector.component.ts [hunk 5]
```diff
@@ -20,6 +20,7 @@ const MIN_DISTANCE_PX = 70;
 })
 export class TuiTableSizeSelectorComponent {
     @Output()
+    // eslint-disable-next-line @angular-eslint/no-output-on-prefix
     readonly onSelectSize = new EventEmitter<{cols: number; rows: number}>();
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/addon-editor/components/toolbar-tools/table-size-selector/table-size-selector.component.ts]: fixed

File: projects/demo/src/modules/components/scrollbar/scrollbar.component.ts [hunk 6]
```diff
@@ -1,9 +1,10 @@
-import {Component} from '@angular/core';
+import {ChangeDetectionStrategy, Component} from '@angular/core';
 import {TuiDocExample} from '@taiga-ui/addon-doc';
 
 @Component({
     selector: 'example-tui-scrollbar',
     templateUrl: './scrollbar.template.html',
+    changeDetection: ChangeDetectionStrategy.OnPush,
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/demo/src/modules/components/scrollbar/scrollbar.component.ts]: fixed

File: projects/addon-editor/components/toolbar-tools/table-size-selector/table-size-selector.component.ts [hunk 7]
```diff
@@ -20,7 +20,15 @@ const MIN_DISTANCE_PX = 70;
...
-    readonly onSelectSize = new EventEmitter<{cols: number; rows: number}>();
+    readonly selectSize = new EventEmitter<{cols: number; rows: number}>();
+
+    /**
+     * @deprecated use {@link selectSize}
+     * TODO: remove in v4.0
+     */
+    @Output()
+    // eslint-disable-next-line @angular-eslint/no-output-on-prefix
+    readonly onSelectSize = this.selectSize;
```
Comment (waterplea) [inline projects/addon-editor/components/toolbar-tools/table-size-selector/table-size-selector.component.ts:31]: Should also update `onClick` method in this component.

File: projects/demo/src/modules/components/input-time/examples/1/index.html [hunk 8]
```diff
@@ -22,7 +22,7 @@
         directive
     </p>
     <tui-input-time
-        tuiUnfinishedValidator
+        tuiUnfinishedValidator="hello"
```
Comment (waterplea) [inline projects/demo/src/modules/components/input-time/examples/1/index.html:25]: Let's use some realistic text.

**Review by waterplea — APPROVED** [review]

File: projects/demo/src/modules/components/input-time/examples/1/index.html (hunk 8 above)
Comment (splincode) [reply projects/demo/src/modules/components/input-time/examples/1/index.html:25]: sorry for that ((

Comment (splincode) [reply projects/demo/src/modules/components/input-time/examples/1/index.html]: fixed

File: projects/addon-editor/components/toolbar-tools/table-size-selector/table-size-selector.component.ts (hunk 7 above)
Comment (splincode) [reply projects/addon-editor/components/toolbar-tools/table-size-selector/table-size-selector.component.ts:31]: fixed

**Pull request merged**